### PR TITLE
COMMERCE-12579

### PR DESCRIPTION
**Pull request by brianikim**
https://liferay.atlassian.net/browse/COMMERCE-12579

@andrea-ale-sbarra

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/helper/CPContentHelperImpl.java [hunk 1]
```diff
@@ -705,6 +680,32 @@ public boolean hasReplacement(
 		return false;
 	}
 
+	@Override
+	public boolean isDirectReplacement(CPSku cpSku) throws Exception {
+		if ((cpSku == null) || cpSku.isDiscontinued()) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/helper/CPContentHelperImpl.java:685]: a discontinued sku may be a replacement for another sku

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/helper/CPContentHelperImpl.java:685]:
```suggestion
		if (cpSku == null) {
```

File: modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_detail/render/view.jsp [hunk 2]
```diff
@@ -254,7 +254,7 @@ String navSpecificationsId = liferayPortletResponse.getNamespace() + "navSpecifi
 			).put(
 				"hasDescription", hasDescription
 			).put(
-				"hasReplacements", hasDirectReplacement
+				"hasReplacements", directReplacement
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_detail/render/view.jsp:257]: I would also rename the key hasReplacement for consistency

File: modules/apps/commerce/commerce-product-content-api/bnd.bnd [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Commerce Product Content API
 Bundle-SymbolicName: com.liferay.commerce.product.content.api
-Bundle-Version: 11.0.1
+Bundle-Version: 12.0.0
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-api/bnd.bnd:3]: this breaking change needs to be documented with a commit description
Reference: https://liferay.atlassian.net/wiki/spaces/ENGR/pages/2489483265/Process+to+register+breaking+changes+in+commits
Example:
https://github.com/brianchandotcom/liferay-portal/pull/142322/commits/d5dde9bfd099fff1cf8f211ebd931ae1377e7fe4

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Overall this looks really good.
Just few changes asked.
Thank you!

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/helper/CPContentHelperImpl.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/helper/CPContentHelperImpl.java:685]: @andrea-ale-sbarra , the method is only being used in the JSP that handles whether the replacementTab should show.
Do we still want to show the replacement tab and all of the replacements even if the SKU is discontinued?

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/helper/CPContentHelperImpl.java:685]: Yes, why not? In the API I believe we are returning these information, so I believe we should do the same for the UI. Anyway this is more a question for the product, do you mind ask the same to David?
Thank you!

**Comment by brianikim** [comment]
@andrea-ale-sbarra ,

I made a straight-forward integration test as you requested and made all the changes requested as of well.

**Comment by brianikim** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
Add automation and resent on https://github.com/liferay-commerce/liferay-portal/pull/4400

**Comment by jaydawu** [comment]
ci:close

**Pull request closed without merging**